crates/subspace-farmer/src/bin/subspace-farmer/main.rs

```diff
@@ -72,13 +75,20 @@ enum Command {
         /// Host and port where built-in WebSocket RPC server should listen for incoming connections
         #[clap(long, short, default_value = "127.0.0.1:9955")]
         ws_server_listen_addr: SocketAddr,
+        /// Address for farming rewards
+        #[clap(long, parse(try_from_str = parse_reward_address))]
+        reward_address: Option<PublicKey>,
     },
 }
 
+fn parse_reward_address(s: &str) -> Result<PublicKey, PublicError> {
+    sp_core::sr25519::Public::from_str(s).map(|key| PublicKey::from(key.0))
```

Comment:
This is a more idiomatic version and doesn't require importing of `std::str::FromStr`:
```suggestion
    s.parse::<sp_core::sr25519::Public>().map(|key| PublicKey::from(key.0))
```